### ISSUE TITLE
`Layer._display_bounding_box()` returns too few dimensions

### ISSUE THREAD

**Issue by brisvag** [napari/napari#4900]
## 🐛 Bug

In working on #4849 I had issues because I assumed I would get consistent output from `Layer._display_bounding_box()`. I would expect a 2d bounding box when passing 2 dimensions, and 3d when passing 3. However, if the layer is a 2D image, I only get 2 dimensions. But I would expect these two layers to behave identically:

```py
Image(np.ones(1, 10, 10))
Image(np.ones(10, 10))
```

@kevinyamauchi @alisterburt am I wrong about my assumptions here?

**Comment by alisterburt** [comment]
Hm, I don't quite see why you would expect the 3D data to give a 2D bounding box unless the image layer drops the dimensionality?

**Comment by brisvag** [comment]
The opposite acually: I would expect to get a 3D bounding box if I ask for 3D:
```py
import napari
import numpy as np

v = napari.Viewer(ndisplay=3)
l1 = v.add_image(np.ones((1, 100, 100)))
l2 = v.add_image(np.ones((100, 100)))

l1._display_bounding_box(l1._dims_displayed)  # get 3 bounds
l2._display_bounding_box(l2._dims_displayed)  # get 2 bounds

l2._display_bounding_box((0, 1, 2))  # get error, but would expect the same as l1, since they *look* the same on the canvas
```

But I'm not sure I agree with myself here xD

**Comment by alisterburt** [comment]
so you currently get a 3D bounding box for 3D data and a 2D one for 2D data?

The `_display_*` methods slice the nD object (in this case bounding box) by the dims in the same way as the data are (currently) sliced... we know there are lots of problems with the way this slicing occurs, tidying this up is I think also part of Andys future work plans

You can think of this as the bounds of the data in view :-) does it at least make sense now?

**Comment by brisvag** [comment]
> You can think of this as the bounds of the data in view :-)

But that's not relaly true, since the data in view for the two layers above is exactly the same, and yet the bounds look different... It's really the bounds for the data in view, except for the dimensions in which that data has implied thickness of 1 :P

Anyways, this is not urgent, just something interesting I noticed... Definitely will be nice to clean up this.

**Comment by alisterburt** [comment]
I see your point but idk, to me it would seem much weirder if the extents of the sliced data had different dimensionality than the data itself 🤷‍♂️ 

Maybe we could have a system where data is explicitly promoted to have ndim=max(layer.data.ndim for layer in viewer.layers) internally prior to slicing? This would likely resolve lots of weirdness like this

**Comment by Czaki** [comment]
So there I a need to move it outside Layer as layer dhould not need to know viewer state for proper working and creating bounding box base on viewer dimensionality break this.

**Comment by alisterburt** [comment]
Agreed @Czaki - I wasn't suggesting literally that we perform this calculation on the layer, rather that the slicing machinery could leave the necessary info somewhere accessible :)

**Comment by brisvag** [comment]
Why would you need info about the layer? You can simply return `(0, 0)` (translated as appropriate) or something like that for any dimension that's not in the data.

**Comment by Czaki** [comment]
And how many of them? This is determined by other layers.

**Comment by brisvag** [comment]
No, it's determined by the argument of `Layer._display_bounding_box`, which is `dims_displayed`. This does not depends on the state of the layer, it simply asks "if you were to be displayed with dimensions `1` and `3`, what would be your bounding box?"
